packages/react-components/react-dialog/src/components/DialogTrigger/useDialogTrigger.ts

```diff
@@ -52,7 +54,7 @@ export const useDialogTrigger_unstable = (props: DialogTriggerProps): DialogTrig
 
   return {
     children: applyTriggerPropsToChildren<DialogTriggerChildProps>(children, {
-      'aria-haspopup': 'dialog',
+      'aria-haspopup': action === 'close' ? undefined : 'dialog',
```

Comment:
We should remove `aria-haspopup` entirely for dialog triggers. In practice, it's only useful for elements that open menus or listboxes, and its future is currently being debated 😅. The `aria-haspopup="dialog"` value mostly announces as if it were opening a menu (i.e. the "dialog" value is ignored), and `aria-expanded` should be enough to indicate that a button or link is opening a dialog.